src/zenml/zen_stores/models/flavor_wrapper.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from typing import Type
+
+from pydantic import BaseModel
+
+from zenml.enums import StackComponentType
+from zenml.stack.stack_component import StackComponent
+from zenml.utils.source_utils import load_source_path_class
+
+
+class FlavorWrapper(BaseModel):
+    """Network serializable wrapper representing the custom implementation of
+    a stack component flavor."""
+
+    name: str
+    integration: str = ""
+    type: StackComponentType
+    source: str
+
```

Comment:
I like the idea, especially with the integration versions. However, time-wise I think it would be more efficient if we conduct it in a follow-up iteration of this PR, for now, we can create a ticket for it in Jira. What do you think?

Reply:
:+1: I was thinking the same, as this will lead to a lot of additional changes.